BirtherDay/Core/Service/CouponService.swift

```diff
@@ -71,6 +72,27 @@ final class CouponService {
         }
     }
     
+    /// 내가 받은 선물 쿠폰 중, 최대 5개까지만 조회합니다.
+    /// /// 내가 받은 선물 쿠폰 중, 최대 5개까지만 조회합니다.
+    /// - Parameter userId: 수신자 ID
+    /// - Returns: 최대 5개의 쿠폰 리스트
+    func retrieveTopFiveReceivedCoupons(_ userId: String) async throws -> PostgrestResponse<[RetrieveCouponResponse]> {
+        do {
+            let res: PostgrestResponse<[RetrieveCouponResponse]> = try await client
+                .from("coupon")
+                .select()
+                .eq("receiver_id", value: userId)
+                .order("deadline", ascending: true)
+                .limit(5)
+                .execute()
+            
+            return res
+        } catch {
+            dump(error)
+            throw CouponError.serverFetchFailed
+        }
+    }
```

Comment:
p2

오오오 직접 코드를 작성하시다니... 멋있습니다 길!
한 가지 피드백이 있자면! 확장성이 별로 좋지 않은 것 같아요!

현재 retrieveTopFiveReceivedCoupons 함수는 조회 개수를 고정값(5개) 으로 설정하고 있는데, 이는 비즈니스 로직에 따라 유동적으로 변경될 수 있는 요소이기 때문에, 하드코딩된 값으로 고정하기보다는 조회할 개수를 외부에서 주입받는 방식이 확장성과 유지보수 측면에서 더 좋다고 생각이 들어용

그래서 retrieveTopFiveReceivedCoupons 함수를 삭제하고 retrieveReceivedCoupons 함수를 수정하는게 좋을 것 같아요오

```suggestion
func retrieveReceivedCoupons(_ userId: String, limit: Int = 5) async throws -> PostgrestResponse<[RetrieveCouponResponse]> {
    guard limit > 0 else {
        throw CouponError.invalidLimit
    }

   do {
         let res: PostgrestResponse<[RetrieveCouponResponse]> = try await client
        .from("coupon")
        .select()
        .eq("receiver_id", value: userId)
        .order("deadline", ascending: true)
        .limit(limit)
        .execute()

       return res
   } catch {
       dump(error)
       throw CouponError.serverFetchFailed
   }
}
```

Reply:
P3
프레이 코드를 보다가 든 생각인데, 길이 요청 코드를 새로 작성한 이유는 Response 받은 쿠폰을 Deadline 순으로 받기 위해서 다시 작성한거라고 생각했어요

그래서 확장성을 고려한다면 조회할 개수를 받는 것 보다 데이터 필터링 종류를 enum을 만들어서 관리하는건 어떤가요 ??
쿠폰 데이터를 필터링 목적에 맞게 받아오기만 한다면 해당 데이터에서 가져오는 개수를 정하는건 데이터를 사용하는 곳에서 처리할 수 있다고 생각이 들었어요

Reply:
p3
enum으로 필터링 종류를 관리하는 것도 좋네요!